### PR TITLE
Russell-L252.MACC

### PR DESCRIPTION
**Pull request by russellhz**
#666 

Waiting on #672 

I added a table for matching in the Guus Velder data.

### CODE REVIEW

File: R/zchunk_L252.MACC.R [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+    L211.AGRBio <- get_data(all_data, "L211.AGRBio")
+    L232.nonco2_prc <- get_data(all_data, "L232.nonco2_prc")
+    L241.hfc_all <- get_data(all_data, "L241.hfc_all")
+    L241.pfc_all <- get_data(all_data, "L241.pfc_all")
+
+    # ===================================================
+    # Prepare the table with all MAC curves for matching
+    # This contains all tax and mac.reduction values
+    L252.MAC_pct_R_S_Proc_EPA <- L152.MAC_pct_R_S_Proc_EPA %>%
+      gather(tax, mac.reduction, matches("[0-9]+")) %>%
```
Comment (bpbond) [inline R/zchunk_L252.MACC.R]: Perhaps change to `"^[0-9]+$"` to be more precise.

File: R/zchunk_L252.MACC.R [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+      mutate(tax = as.numeric(tax)) %>%
+      rename(mac.control = Process)
+
+    MAC_taxes <- unique(L252.MAC_pct_R_S_Proc_EPA$tax)
+
+    # This is a function to add in the mac.reduction curves to data
+    # Function needed because these steps are repeated 5 times
+    mac_reduction_adder <- function(df, order, error_no_match = T){
```
Comment (bpbond) [inline R/zchunk_L252.MACC.R]: `= TRUE) {`

File: R/zchunk_L252.MACC.R [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+        df <- df %>%
+          left_join_error_no_match(L252.MAC_pct_R_S_Proc_EPA, by = c("EPA_region", "mac.control", "tax"))  %>%
+          mutate(mac.reduction = round(mac.reduction, 3))
+      } else {
+        # There are times where the data does not match, so using left_join is necessary
+        df <- df %>%
+          left_join(L252.MAC_pct_R_S_Proc_EPA, by = c("EPA_region", "mac.control", "tax"))  %>%
+                      mutate(mac.reduction = round(mac.reduction, 3))
+      }
+      return(df)
```
Comment (bpbond) [inline R/zchunk_L252.MACC.R]: Could remove this line and change above lines to `df %>% left_join...`

File: R/zchunk_L252.MACC.R [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+      filter(year == min(.$year)) %>%
+      # Add in mac.control
+      # Using left_join b/c mac.control for "other industrial processes" is NA
+      left_join(GCAM_sector_tech %>%
+                                 select(mac.control = EPA_MACC_Sector, supplysector, subsector, stub.technology),
+                               by = c("supplysector", "subsector", "stub.technology")) %>%
+      mac_reduction_adder(order = c("supplysector", "subsector", "stub.technology", "Non.CO2"),
+                          # error_no_match is F, which means we use left_join(L252.MAC_pct_R_S_Proc_EPA)
+                          # because not all mac.controls and regions in L252.MAC_pct_R_S_Proc_EPA
+                          error_no_match = F) %>%
```
Comment (bpbond) [inline R/zchunk_L252.MACC.R]: Here and above, `FALSE`

File: R/zchunk_L252.MACC.R [hunk 5]
```diff
@@ -0,0 +1,394 @@
...
+      na.omit() %>%
+      select(region, supplysector, subsector, stub.technology, year, Non.CO2, mac.control, tax, mac.reduction, EPA_region)
+
+
+    # If we want to use Guus Velders data to replace ours
```
Comment (bpbond) [inline R/zchunk_L252.MACC.R]: ?? Clarify comment please

File: R/zchunk_L2521.MACC_IRR.R [hunk 6]
```diff
@@ -36,8 +36,8 @@ module_emissions_L2521.MACC_IRR <- function(command, ...) {
     # Load required inputs
     A_MACC_TechChange <- get_data(all_data, "emissions/A_MACC_TechChange") %>%
       rename(tech.change = tech_change)
-    L252.AgMAC <- get_data(all_data, "temp-data-inject/L252.AgMAC")
-    L252.MAC_an <- get_data(all_data, "temp-data-inject/L252.MAC_an")
```
Comment (bpbond) [inline R/zchunk_L2521.MACC_IRR.R]: Thanks for doing this. Did you remove corresponding files from `temp-data-inject`?

File: R/zchunk_batch_MACC_TC_SSP.xml.R [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+#' Construct XML data structure for all the \code{MACC_TC_SSP.xml} files.
+#'
+#' @param command API command to execute
+#' @param ... other optional parameters, depending on command
+#' @return Depends on \code{command}: either a vector of required inputs,
+#' a vector of output names, or (if \code{command} is "MAKE") all
+#' the generated outputs: \code{MACC_TC_SSP1.xml}, \code{MACC_TC_SSP2.xml}, and \code{MACC_TC_SSP5.xml}.
+module_emissions_batch_MACC_TC_SSP.xml <- function(command, ...) {
+
+  SSP_NUMS <- c(1,2,5)
```
Comment (bpbond) [inline R/zchunk_batch_MACC_TC_SSP.xml.R]: Spaces after commas please

File: R/zchunk_L252.MACC.R (hunk 3 above)
Comment (russellhz) [reply R/zchunk_L252.MACC.R]: Not sure what you're asking for here. Do you just want to compress lines 103 and 104 into one line?

Comment (bpbond) [reply R/zchunk_L252.MACC.R]: I just meant there's no need for an explicit `return`. But it's fine to leave.

File: R/zchunk_L2521.MACC_IRR.R (hunk 6 above)
Comment (russellhz) [reply R/zchunk_L2521.MACC_IRR.R]: Done

**Pull request merged**